purchases/src/main/kotlin/com/revenuecat/purchases/common/FileHelper.kt

```diff
@@ -25,29 +34,26 @@ internal class FileHelper(
         return file.delete()
     }
 
-    fun readFilePerLines(filePath: String): List<String> {
-        val readLines = mutableListOf<String>()
-        val file = getFileInFilesDir(filePath)
-        FileInputStream(file).use { fileInputStream ->
-            InputStreamReader(fileInputStream).use { inputStreamReader ->
-                BufferedReader(inputStreamReader).use { bufferedReader ->
-                    readLines.addAll(bufferedReader.readLines())
-                }
-            }
+    // This is using a lambda with a Stream instead of returning the Stream itself. This is so we keep
+    // the responsibility of closing the bufferedReader to this class. Note that the Stream should
+    // be used synchronously, otherwise the bufferedReader will be closed before the stream is used.
```

Comment:
This is something that I don't fully like... But I didn't want to leak the responsibility of closing the readers.

Reply:
It makes sense tho